### ISSUE TITLE
Support the newer scope-limited Cloudflare API Tokens, not just API Keys

### ISSUE THREAD

**Issue by mrw** [putyourlightson/craft-blitz#263]
### Is your feature request related to a problem? Please describe.

Blitz currently only support Cloudflare's API Keys, which are tied to an account email address and give access to the full Cloudflare account settings. Until recently this was the only kind of authentication Cloudflare's API supported, however, [Cloudflare now has `API Tokens`](https://support.cloudflare.com/hc/en-us/articles/200167836-Managing-API-Tokens-and-Keys). API Tokens are scope limited and so are safer to use than the older API Keys. 

All API requests are the same between API Tokens and API Keys; only the auth headers change.

### Describe the solution you would like

Blitz [authenticates to the Cloudflare API in this block of code](https://github.com/putyourlightson/craft-blitz/blob/b6e1917ca6fc933b572805c073889d4a1ed774b9/src/drivers/purgers/CloudflarePurger.php#L185-L196):

https://github.com/putyourlightson/craft-blitz/blob/b6e1917ca6fc933b572805c073889d4a1ed774b9/src/drivers/purgers/CloudflarePurger.php#L185-L196

To switch to using API Tokens, you just [pass an `Authorization: Bearer` header instead of the `X-Auth` headers](https://api.cloudflare.com/#getting-started-requests): 

```php
$client = Craft::createGuzzleClient([
    'base_uri' => self::API_ENDPOINT,
    'headers'  => [
        'Content-Type' => 'application/json',
        'Authorization' => 'Bearer ' . Craft::parseEnv($this->apiToken), 
    ]
]);
```

An easy switch for Blitz might just be using the same apiKey field but sending it as a bearer token if the email is empty, or the current Auth-Key method if email is filled in.

**Comment by bencroker** [comment]
Added in https://github.com/putyourlightson/craft-blitz/commit/1d63fd8b2925b8380e44b5dc524a3a11836e8ecf for the next release, thanks Matt!

**Comment by bencroker** [comment]
Released in 3.6.12.
